tests/app/shared/issue/description/description.component.spec.ts

```diff
@@ -0,0 +1,96 @@
+import { of } from 'rxjs';
+
+describe('DescriptionComponent', () => {
+  let descriptionComponent: DescriptionComponent;
+  let issueService: any;
+  let phaseService: PhaseService;
+  let formBuilder: FormBuilder;
+  let thisIssue: Issue;
+  let dialog: any;
+  let errorHandlingService: any;
+
+  beforeEach(() => {
+    formBuilder = new FormBuilder();
+    phaseService = new PhaseService(null, null, null, null, null);
+    phaseService.currentPhase = Phase.phaseTeamResponse;
+
+    dialog = jasmine.createSpyObj('MatDialog', ['open']);
+    errorHandlingService = jasmine.createSpyObj('ErrorHandlingService', ['handleError']);
+    issueService = jasmine.createSpyObj('IssueService', ['getLatestIssue', 'updateIssue']);
+
+    descriptionComponent = new DescriptionComponent(issueService, formBuilder, errorHandlingService, dialog, phaseService, null);
+    thisIssue = Issue.createPhaseBugReportingIssue(ISSUE_WITH_EMPTY_DESCRIPTION);
+    descriptionComponent.issue = thisIssue;
+  });
+
+  it('should be initialised with an issueDescriptionForm', () => {
+    descriptionComponent.ngOnInit();
+    expect(descriptionComponent.issueDescriptionForm.value).toEqual({ description: '' });
+  });
+
+  it('should be updated with correct flag and form value when in edit mode', () => {
+    const descriptionComponentEditState = spyOn(descriptionComponent.changeEditState, 'emit');
+
+    descriptionComponent.ngOnInit();
+    descriptionComponent.changeToEditMode();
+    expect(descriptionComponentEditState).toHaveBeenCalledTimes(1);
+    expect(descriptionComponent.issueDescriptionForm.value).toEqual({ description: thisIssue.description });
+  });
+
+  it('should not have its value updated with issue description is invalid', () => {
+    thisIssue.description = undefined;
+    descriptionComponent.issue = thisIssue;
+    descriptionComponent.ngOnInit();
+    descriptionComponent.changeToEditMode();
+    expect(descriptionComponent.issueDescriptionForm.value).toEqual({ description: '' });
+  });
+
+  it('should returnan error in event of a conflict', () => {
+    // Simulation of getting updated issue from Github.
+    const updatedIssue = thisIssue.clone(phaseService.currentPhase);
+    updatedIssue.description = 'Sample Text';
+    issueService.issues = [];
+    issueService.issues[updatedIssue.id] = updatedIssue;
+
+    const newDescriptionComponent = new DescriptionComponent(issueService, formBuilder, errorHandlingService, dialog, phaseService, null);
+    newDescriptionComponent.issue = thisIssue;
+
+    const viewChangesCaLL = spyOn(newDescriptionComponent, 'viewChanges');
+
+    const form = new NgForm([], []);
+    newDescriptionComponent.ngOnInit();
+    newDescriptionComponent.changeToEditMode();
+
+    issueService.getLatestIssue.and.callFake((x: number) => of(updatedIssue));
+    dialog.open.and.callFake((x: any) => {});
+    errorHandlingService.handleError.and.callFake((x: any) => {});
+    newDescriptionComponent.updateDescription(form);
```

Comment:
May I clarify why a new description component is needed?
Can't we call `ngOnInit`, `changeToEditMode`, and `updateDescription` using `descriptionComponent` ?
(i.e. perform this test using the description component set up in the `beforeEach` function )

Reply:
Yea, I have just realised that... 😅 
I did not think through this clearly.. Thanks you for pointing this out!